### ISSUE TITLE
Failure to correctly decode limits in a GDAL created nc file

### ISSUE THREAD

**Issue by joa-quim** [GenericMappingTools/gmt#2589]
Below, *x* is correct but *y* is not (should be equal to *x*
```
junk.nc: x_min: -3.46944695195e-18 x_max: 10 x_inc: 0.05 name: lon n_columns: 200
junk.nc: y_min: 0.05 y_max: 9.95 y_inc: 0.0495 name: lat n_rows: 200
```

```
$ grdinfo junk.nc -V
grdinfo [WARNING]: No range attribute, guessing registration to be pixel
grdinfo [WARNING]: netCDF grid junk.nc information has zmin = zmax = NaN. Reset to 0/0.
junk.nc: Title: GDAL Band Number 1
junk.nc: Command: Thu Jan 30 00:36:09 2020: GDAL Create( junk.nc, ... )
junk.nc: Remark:
junk.nc: Pixel node registration used [Cartesian grid]
junk.nc: Grid file format: nf = GMT netCDF format (32-bit float), CF-1.7
junk.nc: x_min: -3.46944695195e-18 x_max: 10 x_inc: 0.05 name: lon n_columns: 200
junk.nc: y_min: 0.05 y_max: 9.95 y_inc: 0.0495 name: lat n_rows: 200
junk.nc: z_min: 0 z_max: 0 name: GDAL Band Number 1
junk.nc: scale_factor: 1 add_offset: 0
junk.nc: format: classic
```

[junk.nc.zip](https://github.com/GenericMappingTools/gmt/files/4131791/junk.nc.zip)

**Comment by joa-quim** [comment]
#2590 fixed the *info* issue but there is is still a problem. Plotting the example file shows that it's upside-down. For reference, the ``nearneighbor/nat_nn.ps`` shows the correct thing.
The above issue fix is likely incomplete. It misses a ``HH->row_order = k_nc_start_north;`` but adding it doesn't solve this UD flip error.

![GMTjl_tmp](https://user-images.githubusercontent.com/537321/73484159-b2ad1900-4398-11ea-8a87-107ea67d1400.png)

**Comment by joa-quim** [comment]
Fixed in #2594
